File: ELOAM-App/equipment-service/src/main/java/ELOAM/equipmentservice/services/EquipmentService.java
package ELOAM.equipmentservice.services;

import ELOAM.equipmentservice.entities.EquipmentEntity;
import ELOAM.equipmentservice.repositories.EquipmentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EquipmentService {

    @Autowired
    EquipmentRepository equipmentRepository;

    // Find equipment by its ID
    public EquipmentEntity findEquipmentByID(Long equipmentID) {
        return equipmentRepository.findEquipmentByID(equipmentID);
    }

    // Find all equipment
    public List<EquipmentEntity> findAllEquipment() {
        return (List<EquipmentEntity>) equipmentRepository.findAll();
    }

    // Save equipment to the database
    public void saveEquipment(EquipmentEntity equipmentEntity) {
        equipmentRepository.save(equipmentEntity);
    }

    // Find a list of equipment by their availability status
    public List<EquipmentEntity> findEquipmentByAvailability(Boolean equipmentAvailability) {
        return equipmentRepository.findEquipmentByAvailability(equipmentAvailability);
    }

    // Find a list of equipment by their brand
    public List<EquipmentEntity> findEquipmentByBrand(String equipmentBrand) {
        return equipmentRepository.findEquipmentByBrand(equipmentBrand);
    }

    // Change the availability status of equipment
    public void changeAvailabilityStatus(Long equipmentID, Boolean equipmentAvailability) {
        // Find the equipment by its ID
        EquipmentEntity equipmentEntity = equipmentRepository.findEquipmentByID(equipmentID);
        // Change the availability status
        equipmentEntity.setEquipmentAvailability(equipmentAvailability);
        // Save the changes to the database
        equipmentRepository.save(equipmentEntity);
    }

    // Delete equipment from the database
    public void deleteEquipment(Long equipmentID) {
        equipmentRepository.deleteById(equipmentID);
    }

}

File: ELOAM-App/eloam-frontend-service/src/services/EquipmentService.js
import axios from "axios";

const EQUIPMENT_API_URL = "http://localhost:8083/equipment";

class EquipmentService {
  // Get a list of all equipment
  getAllEquipment() {
    return axios.get(EQUIPMENT_API_URL);
  }

  // Get equipment by its code
  getEquipmentByCode(equipmentCode) {
    return axios.get(EQUIPMENT_API_URL + "/" + equipmentCode);
  }

  // Find a list of equipment by their availability status
  getEquipmentByAvailability(availabilityStatus) {
    return axios.get(EQUIPMENT_API_URL + "/availability/" + availabilityStatus);
  }

  // Find a list of equipment by their brand
  getEquipmentByBrand(equipmentBrand) {
    return axios.get(EQUIPMENT_API_URL + "/brand/" + equipmentBrand);
  }

  // Change the availability status of equipment
  changeEquipmentAvailability(equipmentCode) {
    return axios.get(
      EQUIPMENT_API_URL + "/change-availability/" + equipmentCode
    );
  }

  // Delete equipment from the database
  deleteEquipment(equipmentCode) {
    return axios.delete(EQUIPMENT_API_URL + "/delete/" + equipmentCode);
  }

  // Save equipment to the database
  saveEquipment(equipment) {
    return axios.post(EQUIPMENT_API_URL, equipment);
  }
}

// eslint-disable-next-line import/no-anonymous-default-export
export default new EquipmentService();
